src/mono/wasm/Wasm.Build.Tests/DiagnosticsTests.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Text.RegularExpressions;
+using Xunit.Abstractions;
+using Xunit;
+
+#nullable enable
+
+namespace Wasm.Build.Tests;
+
+public class DiagnosticsTests : WasmTemplateTestsBase
+{
+    public DiagnosticsTests(ITestOutputHelper output, SharedBuildPerTestClassFixture buildContext)
+        : base(output, buildContext)
+    {
+    }
+
+    [Fact]
+    public async Task RunSimpleAppWithProfiler()
```

Comment:
I moved this test from `MemoryTests` class and modified it to validate `dotnet.diag.js` via `FeaturePerfTracing: true` below